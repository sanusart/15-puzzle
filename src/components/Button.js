import React from 'react';
import { Link } from 'react-router';
import PropTypes from 'prop-types';

import './Button.css';

const Button = ({ to, className, onClick, text }) => {
  if (to) {
    return (
      <Link to={ to } className={ className } onClick={ onClick }>
        {text}
      </Link>
    );
  }

  return (
    <div className={ className } onClick={ onClick } role="button">
      {text}
    </div>
  );
};

Button.propTypes = {
  to: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func,
  text: PropTypes.string
};

export default Button;
