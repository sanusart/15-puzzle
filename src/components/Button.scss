@import '../vars.scss';

.btn-start {
  border: 1px solid $color-secondary;
  color: $color-secondary;
  padding: 5px 10px;
  margin: 20px 0 10px 0;
  line-height: 21px;
  height: 21px;
  border-radius: 3px;
  cursor: pointer;
  display: inline-block;
  text-decoration: none;
  &.fill,
  &:hover {
	background: $color-secondary;
	color:#fff;
  }
}
