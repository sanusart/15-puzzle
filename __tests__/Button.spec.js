import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button';

it('Snapshot for <Button /> should match', () => {
  const regularBtn = renderer.create(<Button className="btn-start" text="Text" />);
  const json = regularBtn.toJSON();
  expect(json).toMatchSnapshot();
});

it('Snapshot for <Button /> with link should match', () => {
  const buttonWithLink = renderer.create(<Button className="btn-start" to="/404" text="Text" />);
  const json = buttonWithLink.toJSON();
  expect(json).toMatchSnapshot();
});
