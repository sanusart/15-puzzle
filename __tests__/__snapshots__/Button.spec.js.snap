exports[`test Snapshot for <Button /> should match 1`] = `
<a
  className="btn-start"
  onClick={undefined}>
  Text
</a>
`;

exports[`test Snapshot for <Button /> with link should match 1`] = `
<a
  className="btn-start"
  onClick={[Function]}
  style={Object {}}>
  Text
</a>
`;
